client/packages/system/src/Vaccination/Components/SelectItemAndBatch.tsx

```diff
@@ -1,105 +1,74 @@
+import {
+  VaccinationCourseDoseFragment,
+  VaccinationDetailFragment,
+} from '../api/operations.generated';
 import { SelectBatch } from './SelectBatch';
+import { SelectItem } from './SelectItem';
+import { getSwitchReason } from './getSwitchReason';
 
 export const SelectItemAndBatch = ({
   draft,
   dose,
-  hasExistingSelectedBatch,
+  vaccination,
   isOtherFacility,
+  givenAtOtherStore,
   updateDraft,
 }: {
   dose: VaccinationCourseDoseFragment;
   draft: VaccinationDraft;
-  hasExistingSelectedBatch: boolean;
+  vaccination?: VaccinationDetailFragment | null;
   isOtherFacility: boolean;
+  givenAtOtherStore: boolean;
   updateDraft: (update: Partial<VaccinationDraft>) => void;
 }) => {
   const t = useTranslation();
 
-  const vaccineItemOptions = useMemo(() => {
-    const options =
-      dose?.vaccineCourse.vaccineCourseItems?.map(item => ({
-        label: item.name,
-        value: item.itemId,
-      })) ?? [];
-
-    // If the vaccine item for the selected stock line has since been deleted
-    // Add to list so user can still see it
-    if (!!draft.stockLine && !options.some(o => o.value === draft.itemId)) {
-      options.push({
-        label: draft.stockLine.itemName ?? '',
-        value: draft.stockLine.itemId,
-      });
-    }
-
-    return options;
-  }, [dose.id]);
-
   const isHistorical = draft.date?.toDateString() !== new Date().toDateString();
 
-  const selectBatch =
-    !isHistorical || draft.createTransactions || hasExistingSelectedBatch;
-
-  // Auto-select if there is only one item (and not already selected)
-  useEffect(() => {
-    if (vaccineItemOptions.length === 1 && !draft.itemId) {
-      updateDraft({ itemId: vaccineItemOptions[0]!.value });
-    }
-  }, [vaccineItemOptions]);
+  const showBatchSelect =
+    !givenAtOtherStore &&
+    !isOtherFacility &&
+    (!isHistorical || draft.createTransactions || vaccination?.stockLine);
 
-  if (!draft.given) {
-    return null;
-  }
-
-  if (isOtherFacility) {
-    return <InfoText>{t('messages.no-transaction-other-facility')}</InfoText>;
-  }
-
-  if (!vaccineItemOptions.length) {
-    return <InfoText>{t('messages.no-vaccine-items-configured')}</InfoText>;
-  }
+  const transactionSwitchReason = getSwitchReason(draft, vaccination);
 
   return (
     <>
-      {selectBatch && (
+      {draft.given && (
         <>
-          <InputWithLabelRow
-            label={t('label.vaccine-item')}
-            Input={
-              <Select
-                clearable
-                options={vaccineItemOptions}
-                value={draft.itemId ?? ''}
-                onChange={e =>
-                  updateDraft({ itemId: e.target.value, stockLine: null })
-                }
-                sx={{ flex: 1 }}
-              />
-            }
+          <SelectItem
+            disabled={givenAtOtherStore}
+            dose={dose}
+            draft={draft}
+            updateDraft={updateDraft}
           />
```

Comment:
Almost feel like there should be explicit button to change stock line (or delegate error to server, or something like this), allowing SelectBatch to be simpler, and just displaying stock line in a different component (when it's been already set)